erpnext/stock/doctype/stock_entry/stock_entry.js

```diff
@@ -227,6 +227,56 @@ erpnext.stock.StockEntry = erpnext.stock.StockController.extend({
 
 	items_on_form_rendered: function(doc, grid_row) {
 		erpnext.setup_serial_no();
+	},
+
+	basic_rate: function(doc, cdt, cdn) {
+		var item = frappe.model.get_doc(cdt, cdn);
+		this.calculate_amount(item);
+	},
+
+	calculate_amount: function(item, via_additional_costs) {
+		item.basic_amount = flt(flt(item.transfer_qty) * flt(item.basic_rate),
+			precision("basic_amount", item));
+
+		if (!via_additional_costs) {
+			this.distribute_additional_costs();
+		}
+
+		// get the new changes
+		item = frappe.model.get_doc(item.doctype, item.name);
+
+		item.amount = flt(item.basic_amount + flt(item.additional_cost),
+			precision("amount", item));
+
+		item.valuation_rate = flt(flt(item.basic_rate)
```

Comment:
Amount and valuation rate should be calculated for all rows, on trigger of basic_rate and qty.